app/models/jupiter_core/solr_services/deferred_faceted_solr_query.rb

```diff
@@ -77,6 +77,8 @@ def sort(attributes, orders = nil)
                                else
                                  # We could not find the order so we switch to default sort direction
                                  sort_direction_index = solr_exporter.default_sort_indexes.index(sort_attribute)
+                                 # If we cant find the default sort_direction_index for this sort_attribute default it.
+                                 sort_direction_index = 0 if sort_direction_index.blank?
```

Comment:
I kinda feel like the comment is redundant to what is already there 
> We could not find the order so we switch to default sort direction

Especially if we make the 0 a constant with a meaningful name.
```suggestion
                                 DEFAULT_SORT_DIRECTION_INDEX = 0  # put this at the top of the class or method
                                 # We could not find the order so we switch to default sort direction
                                 sort_direction_index = solr_exporter.default_sort_indexes.index(sort_attribute) 
                                 sort_direction_index = DEFAULT_SORT_DIRECTION_INDEX if sort_direction_index.blank?
```

Something I didn't think of before is why the solr_exporter.default_sort_indexes.index(sort_attribute) doesn't give the correct value or if it ever would. 
```
solr_exporter.default_sort_indexes
=> ["title_ssi"]
```
This else is assuming that the default index is being used, or no `sort` is specified, which isn't the case here.

Reply:
Agree that the comment just reiterates what the code does instead of explaining more of the why behind it. I don't feel that the constant significantly improves the situation though. The code as written is awkward and introducing another indirection on top of it just doesn't feel like it improves things.

We should probably take the need for comments and the difficulty of writing them as a sign that the code should be more fundamentally re-worked.

How about something like 

```ruby
       criteria[:sort_order] << case
                                when sort_attribute == :score
                                  # When sorting by score it only makes sense to use :desc order from the user
                                  # perspective so we ignore the order if one is given.
                                  :desc
                                when order.present? && POSSIBLE_SORT_ORDERS.include?(order)
                                  order
                                when (index = solr_exporter.default_sort_indexes.index(sort_attribute)).present?
                                  solr_exporter.default_sort_direction[index]
                                else
                                  # if we have been passed a sort attribute that isn't actually set up for sorting, we'll
                                  # just fall back to the first thing we have on hand 
                                  solr_exporter.default_sort_direction.first
                                end
    end

as a more self-explanatory approach?